src/solvers/dgmulti/sbp.jl

```diff
@@ -0,0 +1,239 @@
+
+"""
+    DGMulti(approximation_type::AbstractDerivativeOperator;
+            element_type::AbstractElemShape,
+            surface_flux=flux_central,
+            surface_integral=SurfaceIntegralWeakForm(surface_flux),
+            volume_integral=VolumeIntegralWeakForm(),
+            kwargs...)
+
+Create a summation by parts (SBP) discretization on the given `element_type`
+using a tensor product structure.
+
+For more info, see the documentations of
+[StartUpDG.jl](https://jlchan.github.io/StartUpDG.jl/dev/)
+and
+[SummationByPartsOperators.jl](https://ranocha.de/SummationByPartsOperators.jl/stable/).
+"""
+function DGMulti(approximation_type::AbstractDerivativeOperator;
+                 element_type::AbstractElemShape,
+                 surface_flux=flux_central,
+                 surface_integral=SurfaceIntegralWeakForm(surface_flux),
+                 volume_integral=VolumeIntegralWeakForm(),
+                 kwargs...)
+
+  rd = RefElemData(element_type, approximation_type; kwargs...)
+  return DG(rd, nothing #= mortar =#, surface_integral, volume_integral)
+end
+
+function DGMulti(element_type::AbstractElemShape,
+                 approximation_type::AbstractDerivativeOperator,
+                 volume_integral,
+                 surface_integral;
+                 kwargs...)
+
+  DGMulti(approximation_type, element_type=element_type,
+          surface_integral=surface_integral, volume_integral=volume_integral)
+end
+
+
+
+function construct_1d_operators(D::AbstractDerivativeOperator, tol)
+  # StartUpDG assumes nodes from -1 to +1
+  nodes_1d = collect(grid(D))
+  M = SummationByPartsOperators.mass_matrix(D)
+  if M isa UniformScaling
+    weights_1d = M * ones(Bool, length(nodes_1d))
+  else
+    weights_1d = diag(M)
+  end
+  xmin, xmax = extrema(nodes_1d)
+  factor = 2 / (xmax - xmin)
+  @. nodes_1d = factor * (nodes_1d - xmin) - 1
+  @. weights_1d = factor * weights_1d
+
+  D_1d = droptol!(inv(factor) * sparse(D), tol)
+  I_1d = Diagonal(ones(Bool, length(nodes_1d)))
+
+  return nodes_1d, weights_1d, D_1d, I_1d
+end
+
+
+function StartUpDG.RefElemData(element_type::Line,
+                               D::AbstractDerivativeOperator;
+                               tol = 100*eps())
+
+  approximation_type = D
```

Comment:
I think this is fine, but I'm wondering if there's a reason you use `approximation_type=D` rather than `typeof(D)`?